### PR TITLE
Add -valgrindexe option to paratest.

### PR DESCRIPTION
**Pull request by hildeth**
Before, if you typed -valgrindexe as an option to paratest, it would parse the
flag as -valgrind and run valgrind on the compiler as well as the test program.
This is not what the user expects.

I modified the server and client scripts to honor the "exe" suffix if present,
and in that case run valgrind on only the test program.

### CODE REVIEW

File: util/test/paratest.client [hunk 1]
```diff
@@ -10,7 +10,7 @@
 #  testdir - directory to run start_test on
 #  chplenv - Chapel environment variables
 #  futures-mode - include .futures (0=no, 1=all, 2=futures only, 3=futures with skipifs)
-#  valgrind - run valgrind (0=no, 1=yes)
+#  valgrind - run valgrind (0=no, 1=yes, 2=execs only)
```
Comment (lydia-duncan) [inline util/test/paratest.client]: I would change the "yes" here to "comp and exec" or something similar for clarity.

File: util/test/paratest.server [hunk 2]
```diff
@@ -608,6 +608,9 @@ sub main {
             }
         } elsif (/^-valgrind/) {
             $valgrind = 1;
+            if (/^-valgrindexe/) {
+                $valgrind = 2;
+            }
```
Comment (lydia-duncan) [inline util/test/paratest.server]: This is fine as is, but you could also have an
  "} elsif (/^-valgrindexe/) {"
before the
  "} elsif (/^-valgrind/) {"

**Comment by lydia-duncan** [comment]
Comments are for optional changes, otherwise it looks good.  I take you've ensured that the original valgrind case (which we use for nightly valgrind testing) works as expected?

**Comment by hildeth** [comment]
Further testing revealed a coding error.
Updated argument description per suggestion.

**Pull request merged**